src/gluonts/model/forecast_generator.py

```diff
@@ -241,3 +262,114 @@ def __call__(
                     info=batch["info"][i] if "info" in batch else None,
                 )
             assert i + 1 == len(batch[FieldName.FORECAST_START])
+
+
+class ForecastBatch:
+    @property
+    def batch_size(self) -> int:
+        raise NotImplementedError
+
+    @property
+    def mean(self) -> np.ndarray:
+        raise NotImplementedError
+
+    def quantile(self, q: Union[float, str]) -> np.ndarray:
+        raise NotImplementedError
+
+
+@dataclass
+class DistributionForecastBatch(ForecastBatch):
+    start_date: list
+    item_id: Optional[list]
+    info: Optional[list]
+    distr_output: Any  # TODO fix
+    distr_args: list
+
+    def __post_init__(self):
+        self.distr = self.distr_output.distribution(*self.distr_args)
+
+    def __iter__(self) -> Iterator[Forecast]:  # TODO fix
+        distributions = [
+            self.distr_output.distribution(*u)
+            for u in recursively_zip_arrays(self.distr_args)
+        ]
+
+        for i, distr in enumerate(distributions):
+            yield make_distribution_forecast(
+                distr,
+                start_date=self.start_date[i],
+                item_id=self.item_id[i] if self.item_id is not None else None,
+                info=self.info[i] if self.info is not None else None,
+            )
+
+    @property
+    def mean(self) -> np.ndarray:
+        return to_numpy(self.distr.mean())
+
+    def quantile(self, q: Union[float, str]) -> np.ndarray:
+        return to_numpy(self.distr.quantile(q))
+
+
+@dataclass
+class SampleForecastBatch(ForecastBatch):
+    start_date: list
+    item_id: Optional[list]
+    info: Optional[list]
+    sample_batch: np.ndarray
+
+    def __post_init__(self):
+        self._sorted_samples_value = None
+
+    @property
+    def _sorted_samples(self) -> np.ndarray:
+        if self._sorted_samples_value is None:
+            self._sorted_samples_value = np.sort(self.sample_batch, axis=1)
+        return self._sorted_samples_value
+
+    def __iter__(self) -> Iterator[SampleForecast]:
+        for i, sample in enumerate(self.sample_batch):
+            yield SampleForecast(
+                sample,
+                start_date=self.start_date[i],
+                item_id=self.item_id[i] if self.item_id is not None else None,
+                info=self.info[i] if self.info is not None else None,
+            )
+
+    @property
+    def batch_size(self) -> int:
+        return self.sample_batch.shape[0]
+
+    @property
+    def num_samples(self) -> int:
+        return self.sample_batch.shape[1]
+
+    @property
+    def mean(self) -> np.ndarray:
+        if self._mean is not None:
+            return self._mean
+        else:
+            return np.mean(self.samples, axis=1)
+
+    def quantile(self, q: Union[float, str]) -> np.ndarray:
+        q = Quantile.parse(q).value
+        sample_idx = int(np.round((self.num_samples - 1) * q))
```

Comment:
No interpolation?

Reply:
I’ll make this feature-complete eventually :-)